src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -277,14 +276,39 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
             }
 
             var oldContainingSymbol = _factory.CurrentFunction;
+            var oldInstrumenter = _instrumenter;
             try
             {
                 _factory.CurrentFunction = localFunction;
+                if (localFunction.IsDirectlyExcludedFromCodeCoverage)
+                {
+                    _instrumenter = removeDynamicAnalysisInjectors(oldInstrumenter);
+                }
                 return base.VisitLocalFunctionStatement(node);
             }
             finally
             {
                 _factory.CurrentFunction = oldContainingSymbol;
+                _instrumenter = oldInstrumenter;
+            }
+
+            static Instrumenter removeDynamicAnalysisInjectors(Instrumenter instrumenter)
```

Comment:
>removeDynamicAnalysisInjectors [](start = 32, length = 30)

Consider changing this into a member and calling it in the LocalRewriter's constructor in debug mode to ensure that we can handle all possible flavors of instrumenters with having to have a test with local functions.